lib/tool_shed/webapp/frontend/src/components/RepositoryActions.vue

```diff
@@ -1,10 +1,12 @@
 <script setup lang="ts">
-import { fetcher } from "@/schema"
+import { client } from "@/schema"
 import { notify, notifyOnCatch } from "@/util"
-const resetFetcher = fetcher.path("/api/repositories/{encoded_repository_id}/reset_metadata").method("post").create()
 
 async function resetMetadata() {
-    resetFetcher({ encoded_repository_id: props.repositoryId })
+    client
+        .POST("/api/repositories/{encoded_repository_id}/reset_metadata", {
+            params: { path: { encoded_repository_id: props.repositoryId } },
+        })
         .catch(notifyOnCatch)
```

Comment:
This only works if we keep the middleware approach (which is fine, not saying this needs to change)